CMakeLists.txt

```diff
@@ -53,6 +53,15 @@ option(MATPLOTPP_BUILD_HIGH_RESOLUTION_WORLD_MAP "Compile the high resolution ma
 option(MATPLOTPP_BUILD_FOR_DOCUMENTATION_IMAGES "Bypass show() commands and save figures as .svg at destruction" OFF)
 option(MATPLOTPP_BUILD_EXPERIMENTAL_OPENGL_BACKEND "Compile target with the experimental OpenGL backend" OFF)
 
+# Optional dependencies
+option(MATPLOTPP_WITH_JPEG "Require building with JPEG support" OFF)
+option(MATPLOTPP_WITH_TIFF "Require building with TIFF support" OFF)
+option(MATPLOTPP_WITH_PNG "Require building with PNG support" OFF)
+option(MATPLOTPP_WITH_LAPACK "Require building with LAPACK support" OFF)
+option(MATPLOTPP_WITH_BLAS "Require building with BLAS support" OFF)
+option(MATPLOTPP_WITH_FFTW "Require building with FFTW support" OFF)
+option(MATPLOTPP_WITH_OpenCV "Require building with OpenCV support" OFF)
+
 # Where to find dependencies
 option(MATPLOTPP_WITH_SYSTEM_CIMG "Use system-provided CImg.h instead of bundled" OFF)
 option(MATPLOTPP_WITH_SYSTEM_NODESOUP "Use system-provided nodesoup instead of bundled" OFF)
```

Comment:
This option got here through a contribution but it doesn't make much sense either. Zero people have "system nodesoup" installed. nodesoup was a POC library that happened to be very useful here. It's not being maintained at all and should become part of matplot++. I talked to the author and he recommended the source files are just moved into the matplot++ source files into some `detail` directory and compiled with everything else.